src/home/room_screen.rs

```diff
@@ -1711,6 +1734,13 @@ impl RoomScreen {
                     // if the list of typing users gets updated many times in a row.
                     typing_users = users;
                 }
+
+                TimelineUpdate::CanUserPost(can_user_post) => {
+                    let (bottom_input_visible, no_send_permission_notice_visible) = if can_user_post { (true, false) } else { (false, true) };
+
+                    bottom_input.set_visible(bottom_input_visible);
+                    no_send_permission_notice.set_visible(no_send_permission_notice_visible)
```

Comment:
don't overcomplicate things.
```suggestion
                    bottom_input.set_visible(can_user_post);
                    no_send_permission_notice.set_visible(!can_user_post)
```